model/metrics/v8js-metrics.yaml

```diff
@@ -0,0 +1,96 @@
+groups:
+  - id: metric.veightjs.gc.duration
+    type: metric
+    metric_name: v8js.gc.duration
+    brief: "Garbage collection duration."
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    prefix: v8js.gc
+    attributes:
+      - ref: v8js.gc.type
+        requirement_level: recommended
+
+  - id: metric.veightjs.memory.total
+    type: metric
+    metric_name: v8js.memory.total
+    brief: "Total memory size pre-allocated."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: v8js.memory.type
+        requirement_level: recommended
+      - ref: v8js.heap.space.name
+        requirement_level:
+          conditionally_required: If memory type is `heap`.
+
+  - id: metric.veightjs.memory.used
+    type: metric
+    metric_name: v8js.memory.used
+    brief: "Used memory size."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: v8js.memory.type
+        requirement_level: recommended
+      - ref: v8js.heap.space.name
+        requirement_level:
+          conditionally_required: If memory type is `heap`.
+
+  - id: metric.veightjs.heap.space.size
+    type: metric
+    metric_name: v8js.heap.space.size
+    brief: "Total heap space size pre-allocated."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    prefix: v8js.heap.space.size
+    attributes:
+      - ref: v8js.heap.space.name
+        requirement_level: recommended
+    note: >
+      Value can be retrieved from value `space_size` of [`v8.getHeapSpaceStatistics()`](https://nodejs.org/api/v8.html#v8getheapspacestatistics)
+
+  - id: metric.veightjs.heap.space.used_size
+    type: metric
+    metric_name: v8js.heap.space.used_size
+    brief: "Heap space used size."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    prefix: v8js.heap.space.used_size
+    attributes:
+      - ref: v8js.heap.space.name
+        requirement_level: recommended
+    note: >
+      Value can be retrieved from value `space_used_size` of [`v8.getHeapSpaceStatistics()`](https://nodejs.org/api/v8.html#v8getheapspacestatistics)
+
+  - id: metric.veightjs.heap.space.available_size
+    type: metric
+    metric_name: v8js.heap.space.available_size
```

Comment:
it this metric derivable from other ones?  `v8js.heap.space.size` - `v8js.heap.space.used_size` ?

Reply:
Unfortunately it's not. 
The size is the value pre-allocated that can grow, and the used is the already allocated. The value of available is the total capacity - used, but the heap stats function from v8 doesn't return the capacity, just the allocated size. The value of capacity can probably be found in another place, but that would mean that I need to add a new metric to be able to calculate the available. But since the getHeapSpaceStatistics already returns the available, I'm using that directly